### ISSUE TITLE
coalescing bug in TLAPS

### ISSUE THREAD

**Issue by muenchnerkindl** [tlaplus/tlaplus#40]
The PTL backend of TLAPS can be tricked into believing that an obligation appears in a []-context when in fact it does not. A primed state formula can then be "proved" using PTL when the corresponding unprimed formula is available as an assumption. The following module illustrates the bug.

```
---- MODULE Error ----
EXTENDS Naturals
VARIABLE xyz

Init == xyz = 0
Inc(x) == xyz' = xyz+x
Next == \E x \in Nat : Inc(x)
Spec == Init /\ [][Next]_xyz

LEMMA Spec => []Init
<1>1. Init /\ [Next]_xyz => Init'
  <2>. SUFFICES ASSUME Init, Next PROVE Init'  BY DEF Init
  (* the following step illustrates the error and should not be provable *)
  <2>1. ASSUME NEW x \in Nat, Inc(x) PROVE Init'  BY PTL
  <2>. QED BY <2>1 DEF Next
<1>. QED  BY <1>1, PTL DEF Spec
===============================
```

**Comment by lemmy** [comment]
I set the issue's milestone to 1.5.3 assuming it's something we want to address in/with the upcoming Toolbox release. Please unset if incorrect.

**Comment by quicquid** [comment]
It seems even worse in this case, since the following lemma is accepted by PTL:

    LEMMA ASSUME Init PROVE Init' BY PTL

I assume Init is mistakenly assumed as constant.

**Comment by damiendoligez** [comment]
This is a TLAPM bug and I fixed it in commit e1d3cd0589ef106107381259731bea0e217ac771 (on the gforge repo).

**Comment by lemmy** [comment]
Closing as part of housekeeping, please reopen if necessary.
